build.go

```diff
@@ -0,0 +1,111 @@
+	"github.com/paketo-buildpacks/packit/v2/scribe"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+// Note that Go 1.18 requires faux 0.21.0 (https://github.com/ryanmoran/faux/releases/tag/v0.21.0)
+
+//go:generate faux --interface DependencyManager --output fakes/dependency_manager.go
+type DependencyManager interface {
+	Resolve(path, id, version, stack string) (postal.Dependency, error)
+	Deliver(dependency postal.Dependency, cnbPath, layerPath, platformPath string) error
+}
+
+type EntryResolver interface {
+	Resolve(name string, entries []packit.BuildpackPlanEntry, priorites []interface{}) (packit.BuildpackPlanEntry, []packit.BuildpackPlanEntry)
+	MergeLayerTypes(string, []packit.BuildpackPlanEntry) (launch, build bool)
+}
+
+func Build(
+	logger scribe.Emitter,
+	dependencyManager DependencyManager,
+	entryResolver EntryResolver) packit.BuildFunc {
+	return func(context packit.BuildContext) (packit.BuildResult, error) {
+		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
+		logger.Process("Resolving Composer version")
+
+		priorities := []interface{}{
+			"BP_COMPOSER_VERSION",
+		}
+		entry, sortedEntries := entryResolver.Resolve("composer", context.Plan.Entries, priorities)
+		logger.Candidates(sortedEntries)
+
+		composerLayer, err := context.Layers.Get("composer")
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		composerLayer, err = composerLayer.Reset()
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		composerLayer.Launch, composerLayer.Build = entryResolver.MergeLayerTypes("composer", context.Plan.Entries)
+
+		if !composerLayer.Launch && !composerLayer.Build {
+			composerLayer.Build = true
+		}
+
+		version, ok := entry.Metadata["version"].(string)
+		if !ok {
+			version = "default"
+		}
+
+		dependency, err := dependencyManager.Resolve(
+			filepath.Join(context.CNBPath, "buildpack.toml"),
+			entry.Name,
+			version,
+			context.Stack)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		clock := chronos.DefaultClock
+
+		logger.SelectedDependency(entry, dependency, clock.Now())
+
+		logger.Process("Executing build process")
+		logger.Subprocess("Installing Composer %s", dependency.Version)
+
+		layerBinPath := filepath.Join(composerLayer.Path, "bin")
+		err = os.MkdirAll(layerBinPath, os.ModePerm)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		duration, err := clock.Measure(func() error {
+			return dependencyManager.Deliver(dependency, context.CNBPath, layerBinPath, context.Platform.Path)
+		})
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+		logger.Action("Completed in %s", duration.Round(time.Millisecond))
+		logger.Break()
+
+		fullFilename := filepath.Join(layerBinPath, filepath.Base(dependency.Name))
+
+		logger.Debug.Subprocess("Delivered Composer filename %s", fullFilename)
+
+		err = os.Chmod(fullFilename, 0755)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
```

Comment:
Is this necessary to do? It very well might be just interested to know

Reply:
It's necessary. Without this the permissions are `-rw-r--r--`/`0644`. 

Do we need to toggle the executable bit for Owner/Group/User? Should it just be owner?

Reply:
I think `0700` should be enough yea